recipes/cyrus-sasl/all/conanfile.py

```diff
@@ -0,0 +1,175 @@
+import glob
+import os
+
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class CyrusSaslConan(ConanFile):
+    name = "cyrus-sasl"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.cyrusimap.org/sasl/"
+    description = "This is the Cyrus SASL API implementation. It can be used on the client or server side to provide authentication and authorization services."
+
+    topics = ("SASL", "authentication", "authorization")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openssl": [True, False],
+        "with_cram": [True, False],
+        "with_digest": [True, False],
+        "with_scram": [True, False],
+        "with_otp": [True, False],
+        "with_krb4": [True, False],
+        "with_gssapi": [True, False],
+        "with_plain": [True, False],
+        "with_anon": [True, False],
+        "with_postgresql": [True, False],
+        "with_mysql": [True, False],
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True,
+        "with_openssl": True,
+        "with_cram": True,
+        "with_digest": True,
+        "with_scram": True,
+        "with_otp": True,
+        "with_krb4": True,
+        "with_gssapi": True,
+        "with_plain": True,
+        "with_anon": True,
+        "with_postgresql": False,
+        "with_mysql": False,
+    }
+    exports_sources = ("patches/**",)
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration(
+                "Cyrus SASL package is not compatible with Windows yet."
+            )
+        if self.options.with_gssapi and not self.options.shared:
+            raise ConanInvalidConfiguration(
+                "Cyrus SASL package cannot link statically to libkrb5"
+            )
+
+    def requirements(self):
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1g")
+        if self.options.with_postgresql:
+            self.requires("libpq/12.2")
+        if self.options.with_mysql:
+            self.requires("libmysqlclient/8.0.17")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        downloaded_folder_name = f"{self.name}-{self.version}"
```

Comment:
Python 2 is no longer supported. Why do you want to keep it compatible @uilianries?

Reply:
Conan 1.x promotes backward compatibility, when it started, python 2 was alive, thus breaking py2 now can break some users. It will be remove for Conan 2.x. I also would like to run only py3 for all recipes, but we can't now.